base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1010,16 +1010,17 @@ where
             None,
         )?;
 
-        // Clear any existing pending coinbase transactions for this blockheight if they exist
+        // Clear any existing pending coinbase transactions having this output hash, if they exist
         match self
             .resources
             .db
-            .clear_pending_coinbase_transaction_at_block_height(block_height)
+            .clear_pending_coinbase_transaction_with_hash(output.hash.as_slice())
```

Comment:
I dont think we have to clear this?
Rather than deleting the same utxo, we just see if its the same and retrieve that from the db and don't save a new one. 

Currently if its the same, it will delete the old one and replace it with an exact duplicate.

Reply:
We need to check this on the commitment, and not the complete output. The Rangeproof, and signatures have random nonce's in it. 
As of this PR, the commitment is deterministic though, so for each set of (fee + rewards), it will be unique.

Reply:
Changes to this were done, thanks !